pkg/messaging/nats/pubsub.go

```diff
@@ -42,7 +42,7 @@ type subscription struct {
 }
 
 type pubsub struct {
-	conn          *broker.Conn
+	publisher     publisher
```

Comment:
Instead of having a `publisher` field, consider using a composition of `publisher` like:
```go
type pubsub struct {
	publisher
	logger        log.Logger
	mu            sync.Mutex
	queue         string
	subscriptions map[string]map[string]subscription
}
```
This will enable using `ps.conn` rather than `ps.publisher.conn` later on.

Reply:
That is what we had initially but @drasko had suggested with name it.

I will stick with the earlier changes we had.